src/software/ai/hl/stp/tactic/pass_defender/pass_defender_fsm.cpp

```diff
@@ -105,3 +105,42 @@ void PassDefenderFSM::interceptBall(const Update& event)
         TbotsProto::DribblerMode::MAX_FORCE, TbotsProto::BallCollisionType::ALLOW,
         AutoChipOrKick{AutoChipOrKickMode::OFF, 0}));
 }
+
+bool PassDefenderFSM::ballNearbyWithoutThreat(const Update& event)
+{
+    Point robot_position   = event.common.robot.position();
+    double ball_position_x = event.common.world_ptr->ball().position().x();
+    std::optional<Robot> nearest_enemy =
+            event.common.world_ptr->enemyTeam().getNearestRobot(robot_position);
+    if (nearest_enemy)
+    {
+        // Get the ball if ball is closer to robot than enemy threat by threshold ratio
+        double ball_distance =
+                distance(robot_position, event.common.world_ptr->ball().position());
+        double nearest_enemy_distance =
+                distance(robot_position, nearest_enemy->position());
+
+        return ball_distance < nearest_enemy_distance * MAX_GET_BALL_RATIO_THRESHOLD &&
+               ball_position_x < 0 && ball_distance <= MAX_GET_BALL_RADIUS_M &&
+               event.common.world_ptr->ball().velocity().length() <=
+               MAX_BALL_SPEED_TO_GET_MS;
+    }
+    else
+    {
+        return true;
+    }
+}
+
+void PassDefenderFSM::prepareGetPossession(
+        const Update& event, boost::sml::back::process<DribbleFSM::Update> processEvent)
+{
+    Point ball_position       = event.common.world_ptr->ball().position();
+    Point enemy_goal_center   = event.common.world_ptr->field().enemyGoal().centre();
+    Vector ball_to_net_vector = Vector(enemy_goal_center.x() - ball_position.x(),
+                                       enemy_goal_center.y() - ball_position.y());
```

Comment:
I haven't really try to compile this in your branch so please be careful. Maybe something like the following: 

```
Point ball_position       = event.common.world_ptr->ball().position();
Point enemy_goal_center   = event.common.world_ptr->field().enemyGoal().centre();
Vector ball_to_net_vector = (enemy_goal_center-ball_position).toVector()
```

see here: 

https://github.com/UBC-Thunderbots/Software/blob/99e9e9e2a883d01122a377dbee8a12b7d50d659f/src/software/geom/point.cpp#L42-L45

Reply:
LOL fasted PR review in my life. Yes, I will fix that along with adding the tests. I published this at 2 AM thinking I would finish the PR today :0